include/configs/crocodile.h

```diff
@@ -63,9 +65,16 @@
 	"bootqspi=echo Booting from qspi ...;" \
 		"nor_img_addr="__stringify(CONFIG_LEICA_FSPI_NOR_IMG_ADDR)"; " \
 		"nor_img_size="__stringify(CONFIG_LEICA_FSPI_NOR_IMG_SIZE)"; " \
-		"sf probe; "\
 		"sf read ${initrd_addr} ${nor_img_addr} ${nor_img_size}; "\
-		"bootm ${initrd_addr};\0" \
+		"bootm ${initrd_addr}#${fit_config};\0" \
+	"get_fit_config=sf probe;" \
+		"mw.l ${loadenv_addr} 0x0 0x1;" \
+		"sf read ${loadenv_addr} ${board_id_addr} 0x1;" \
+		"if itest.s *${loadenv_addr} < D; then" \
+		"    env set fit_config conf-freescale_crocodile-pt1.dtb;" \
+		"else" \
+		"    env set fit_config conf-freescale_crocodile-pt1.dtb;" \
+		"fi;\0" \
```

Comment:
If there is no board ID, or if the data is corrupt, nobody will know about it.
We could do something like:
```
" env set fit_config conf-freescale_crocodile-pt1.dtb;" \
"if itest.s *${loadenv_addr} = A; then" \
"    echo \"Board A - select config conf-freescale_crocodile-pt1.dtb\""
"elif itest.s *${loadenv_addr} = B; then" \
"    echo \"Board B - select config conf-freescale_crocodile-BBB.dtb\""
"    env set fit_config conf-freescale_crocodile-BBB.dtb;" \
"else" \
"    echo \"Board ID unknown - falling back to default config conf-freescale_crocodile-pt1.dtb\""
"fi;\0" \

```

Reply:
Ok, so at the moment I would keep the conditions:
- Board A (the only one we have at the moment)
- Board ID unknown

In future we can add more